app/src/docs/_examples/tables/ArrayTable.example.tsx

```diff
@@ -1,20 +1,23 @@
 import React, { useMemo, useState } from 'react';
-import { DataColumnProps, useArrayDataSource } from '@epam/uui-core';
+import { DataColumnProps, useDataRows, useTree } from '@epam/uui-core';
 import { DataTable, Panel, Text } from '@epam/uui';
 import { demoData, FeatureClass } from '@epam/uui-docs';
 import css from './TablesExamples.module.scss';
 
 export default function ArrayDataTableExample() {
-    const [value, onValueChange] = useState({});
+    const [dataSourceState, setDataSourceState] = useState({});
 
-    const dataSource = useArrayDataSource<FeatureClass, number, unknown>(
-        {
-            items: demoData.featureClasses,
-        },
-        [],
-    );
+    const { tree, ...dataRowProps } = useTree({
+        getId: (item) => item.id,
+        dataSourceState,
+        setDataSourceState,
+        items: demoData.featureClasses,
+    }, []);
 
-    const view = dataSource.useView(value, onValueChange, {});
+    const { listProps, visibleRows } = useDataRows({
```

Comment:
Let's try to adjust useDataRows output and DataTable's props to be compatible, and more convenient.

My proposal:
```
const { rowsCount, rows, totalCount } = useDataRows(); // to use separately

// to pass-thru to the table: 
const rowProps = useDataRows();
<DataTable { ...rowsProps } />
```
DataTable needs a minimal adjustments to allow ```rows``` prop, as alternative to getVisibleRows.

I also believe that ```rows``` is better than ```visibleRows```, as 'visible' doesn't add any clarification in this case.

Reply:
I agree on `rows`. It can be aligned without any problem.
There is a problem with the `pass-through` approach.
```ts
interface DataRowsResult<TItem, TId> {
  rows: DataRowProps<TItem, TId>[];
  getSelectedRows: : ({ topIndex, visibleCount }?: VirtualListRange) => DataRowProps<TItem, TId>[];
  getSelectedRowsCount: : () => number;
  getById:  (id: TId, index: number) => DataRowProps<TItem, TId>;
  clearAllChecked: () => void;
        
  listProps: DataSourceListProps;

  selectAll: {
    value: boolean;
    onValueChange: (isChecked: boolean) => void;
    indeterminate: boolean;
    isDisabled?: undefined;
  } | {
    value: boolean;
    onValueChange: () => void;
    isDisabled: boolean;
    indeterminate: boolean;
  },
}
```

This is a return type of `useDataRows` hook. Sometimes, all the properties, returned from this hook, are required. And not all of them should be passed to the `DataTable`. If listProps is spread too, it will turn into a mess. Don't you think that the best idea is to leave it as a separate property, which collects all the stats about the datasource.


Otherwise, we'll face the further issue:
```ts
const {
    getSelectedRows,
    getSelectedRowsCount,
    getById,
    clearAllChecked,
    selectAll,
    ...tableProps,
} = useDataRows(props);

<DataTable { ...tableProps } />;
```
To pass necessary data to a DataTable it will be required to destruct all the other props, which looks weird.
It seems to me, it is much better to extract only rows and listProps:
```ts
const { rows, listProps } = useDataRows(props);

<DataTable rows={ rows } { ...listProps } />;
```

@jakobz, WDYT about it?